File: src/Action/Sound.cpp
//
// Created by Dylan Meadows on 2020-03-16.
//

#include "Sound.h"

namespace EonTimer::Action {
    const std::vector<Sound> &getSounds() {
        static const std::vector<Sound> values{BEEP, DING, TICK, POP};
        return values;
    }

    const QString &getName(const Sound sound) {
        static const std::vector<QString> names{"Beep", "Ding", "Tick", "Pop"};
        return names[sound];
    }
}  // namespace EonTimer

File: src/main/kotlin/io/eontimer/action/Sound.kt
package io.eontimer.action

import io.eontimer.util.BASE_RESOURCE_PATH
import io.eontimer.util.Resource
import io.eontimer.util.javafx.Choice
import java.util.Locale

enum class Sound(
    fileName: String
) : Resource, Choice {
    BEEP("beep.wav"),
    TICK("tick.wav"),
    DING("ding.wav"),
    POP("pop.wav");

    override val path = "$BASE_RESOURCE_PATH/sounds/$fileName"
    override val displayName = name.lowercase().replaceFirstChar {
        if (it.isLowerCase()) it.titlecase(Locale.getDefault()) else it.toString()
    }
}